modules/light-clients/08-wasm/keeper/msg_server_test.go

```diff
@@ -53,11 +56,25 @@ func (suite *KeeperTestSuite) TestMsgStoreCode() {
 			types.ErrWasmCodeExists,
 		},
 		{
-			"fails with invalid wasm code",
+			"fails with zero-length wasm code",
 			func() {
 				msg = types.NewMsgStoreCode(signer, []byte{})
 			},
-			types.ErrWasmEmptyCode,
+			fmt.Errorf("Wasm bytes nil or empty"),
+		},
+		{
+			"fails with checksum",
+			func() {
+				msg = types.NewMsgStoreCode(signer, []byte{0, 1, 3, 4})
+			},
+			errorsmod.Wrap(errorsmod.Wrap(errors.New("Wasm bytes do not not start with Wasm magic number"), "wasm bytecode checksum failed"), "failed to store wasm bytecode"),
```

Comment:
@DimitrisJim Can you please help me here to construct the error correctly so that `ErrorIs` succeeds? I am trying a couple of ways but none work. 😓

Reply:
I can take a quick look if you're busy @DimitrisJim 👍

Reply:
Since we're not importing a specific error directly, and effectively duplicating it, I think we might need to modify our check to 

```golang
suite.Require().Contains(tc.expError.Error(), err.Error())
```

Reply:
also a bit hesitant to duplicate this check as if the error string changes in CosmWasm this test will break

Reply:
it looks like the `wrapError` type which `fmt.Errorf` returns, does not implement the `Is(error)bool` interface which is dynamically looked for in `errors.Is`. Because of this, it seems `errors.Is` will always return false if a `wrappedError` is ultimately passed.

Reply:
Thank you for the help, guys!